### PR TITLE
add dmg result mapping function for galveston

### PR DESCRIPTION
**Pull request by Rashmil-1999**
The diff is big since I have reformatted the file neatly. You only need to look at `create_mapped_dmg`, `create_mapped_dmg_result` and `create_mapped_dmg_result_gal` function where the former 2 are actually unchanged but just have been split up as I needed to repeat that code in the last function. Logic is pretty simple though :p

To test it use this updated [nb.zip](https://github.com/IN-CORE/pyincore/files/14291000/nb.zip) notebook after installing this pyincore of this branch!

-----
Also you can test it with this PR:
https://github.com/IN-CORE/incore-docs/pull/342

### CODE REVIEW

File: pyincore/utils/dataprocessutil.py [hunk 1]
```diff
@@ -283,3 +427,75 @@ def get_max_damage_state(dmg_result):
...
+        max_dmg_result.loc[max_dmg_result["max_state"] == "DS_1", "max_state"] = "DS_0"
+        max_dmg_result.loc[max_dmg_result["max_state"] == "DS_2", "max_state"] = "DS_1"
+        max_dmg_result.loc[
+            (
+                (max_dmg_result["max_state"] == "DS_3")
+                & (max_dmg_result["sw_max_ds"] != "DS_3")
+            ),
+            "max_state",
+        ] = "DS_2"
+        max_dmg_result.loc[max_dmg_result["sw_max_ds"] == "DS_3", "max_state"] = "DS_3"
```
Comment (longshuicy) [inline pyincore/utils/dataprocessutil.py:471]: Would it be easier just do the renaming here and then you don't need line 477-494?
```
 max_dmg_result.loc[max_dmg_result["max_state"] == "DS_1", "max_state"] = "Affected"
        max_dmg_result.loc[max_dmg_result["max_state"] == "DS_2", "max_state"] = "Minor"
        max_dmg_result.loc[
            (
                (max_dmg_result["max_state"] == "DS_3")
                & (max_dmg_result["sw_max_ds"] != "DS_3")
            ),
            "max_state",
        ] = "Major"
        max_dmg_result.loc[max_dmg_result["sw_max_ds"] == "DS_3", "max_state"] = "Destroyed"
```

File: pyincore/utils/dataprocessutil.py [hunk 2]
```diff
@@ -123,55 +154,114 @@ def get_mapped_result_from_path(inventory_path: str, dmg_result_path: str,
...
+            dmg_result: dmg_result that need to merge with inventory and be grouped
+            arch_mapping: Path to the archetype mappings
+
+        Returns:
+            ret_json: JSON of the results ordered by cluster and category.
+
+        """
+        result_by_cluster, result_by_category = DataProcessUtil.create_mapped_dmg(
+            inventory, dmg_result, arch_mapping, groupby_col_name, arch_col
+        )
```
Comment (longshuicy) [inline pyincore/utils/dataprocessutil.py:264]: Could we also do the renaming here; so frontend can just directly render the column header as it is
```
result_by_cluster.rename(
            columns={
                "DS_0": "Insignificant",
                "DS_1": "Moderate",
                "DS_2": "Extensive",
                "DS_3": "Complete",
            },
            inplace=True,
        )
        result_by_category.rename(
            columns={
                "DS_0": "Insignificant",
                "DS_1": "Moderate",
                "DS_2": "Extensive",
                "DS_3": "Complete",
            },
            inplace=True,
        )
```

Comment (longshuicy) [reply pyincore/utils/dataprocessutil.py:264]: I created another frontend task related to this
https://github.com/IN-CORE/incore-playbook/issues/285
If you also want to work on that, feel free to work on that; I put it in the next week's sprint. :-)

**Review by longshuicy — CHANGES_REQUESTED** [review]

File: pyincore/utils/dataprocessutil.py (hunk 1 above)
Comment (longshuicy) [reply pyincore/utils/dataprocessutil.py:471]: Ah also don't forget do 
```
max_dmg_result.loc[max_dmg_result["max_state"] == "DS_0", "max_state"] = "Affected"
```
if you are taking this approach

Comment (Rashmil-1999) [reply pyincore/utils/dataprocessutil.py:471]: yes, that shortens stuff... I can get rid of the helper function.....

Comment (Rashmil-1999) [reply pyincore/utils/dataprocessutil.py:471]: Ohh, I think not, I'll still need it...

File: pyincore/utils/dataprocessutil.py (hunk 2 above)
Comment (Rashmil-1999) [reply pyincore/utils/dataprocessutil.py:265]: yep, I can work on it next.

**Review by longshuicy — APPROVED** [review]

**Review by ywkim312 — APPROVED** [review]
Test with galveston community app notebook PR. Worked okay. Approve

**Pull request merged**